src/americana.js

```diff
@@ -255,20 +255,38 @@ function buildLayers() {
     lyrPlace.continent
   );
 
-  // Resolve localized name placeholders.
+  // Updated localizable variables at the top level of each layer's text-field expression based on the current language preference.
+  let locale = Label.getLocales()[0];
   let localizedNameExpression = Label.getLocalizedNameExpression(false);
   let legacyLocalizedNameExpression = Label.getLocalizedNameExpression(true);
   for (let layer of layers) {
-    if (
-      "metadata" in layer &&
-      "layout" in layer &&
-      layer.metadata["americana:text-field-localized"] === true
-    ) {
-      // https://github.com/openmaptiles/openmaptiles/issues/769
-      layer.layout["text-field"] =
-        layer["source-layer"] === "transportation_name"
-          ? legacyLocalizedNameExpression
-          : localizedNameExpression;
+    if ("layout" in layer && "text-field" in layer.layout) {
+      let textField = layer.layout["text-field"];
+      if (textField && textField[0] === "let") {
+        if (textField.indexOf("localizedName") % 2 === 1) {
+          let variableNameIndex = textField.indexOf("localizedName");
+          if (variableNameIndex % 2 === 1) {
+            // https://github.com/openmaptiles/openmaptiles/issues/769
+            let expr =
+              layer["source-layer"] === "transportation_name"
+                ? legacyLocalizedNameExpression
+                : localizedNameExpression;
+            layer.layout["text-field"][variableNameIndex + 1] = expr;
+          }
+        }
+        if (textField.indexOf("localizedCollator") % 2 === 1) {
+          let variableNameIndex = textField.indexOf("localizedCollator");
+          if (variableNameIndex % 2 === 1) {
+            let options = {
+              "case-sensitive": false,
+              "diacritic-sensitive": true,
+              locale: locale,
```

Comment:
Ideally, we’d compare the two strings using both the user-preferred language’s collator and the local language’s collator. Unfortunately, OSM [doesn’t explicitly tag the local language](https://wiki.openstreetmap.org/wiki/Proposed_features/Language_information_for_name) used for `name`. Perhaps the vector tiles could encode the `default_language` of the most immediate administrative boundary. But either approach could be a lot of heft anyways, maybe not worth it for what would probably be quite an edge case.